spawn.lua

```diff
@@ -67,23 +65,18 @@ end
 
 function HandleSetSpawnCommand(Split, Player)
 
-	local WorldIni = cIniFile()
-	WorldIni:ReadFile(Player:GetWorld():GetIniFileName())
+	local World = Player:GetWorld()
 
-	local PlayerX = Player:GetPosX()
-	local PlayerY = Player:GetPosY()
-	local PlayerZ = Player:GetPosZ()
+	local PlayerX = math.floor(Player:GetPosX())
+	local PlayerY = math.floor(Player:GetPosY())
+	local PlayerZ = math.floor(Player:GetPosZ())
 
-	WorldIni:DeleteValue("SpawnPosition", "X")
-	WorldIni:DeleteValue("SpawnPosition", "Y")
-	WorldIni:DeleteValue("SpawnPosition", "Z")
-
-	WorldIni:SetValueI("SpawnPosition", "X", PlayerX)
-	WorldIni:SetValueI("SpawnPosition", "Y", PlayerY)
-	WorldIni:SetValueI("SpawnPosition", "Z", PlayerZ)
-	WorldIni:WriteFile(Player:GetWorld():GetIniFileName())
-
-	SendMessageSuccess(Player, string.format("Changed spawn position to [X:%i Y:%i Z:%i]", PlayerX, PlayerY, PlayerZ))
-	return true
+	if ( World:SetSpawn(PlayerX + 0.5, PlayerY, PlayerZ + 0.5) ) then
```

Comment:
@bearbin Didn't occur to me then
@Bond-009 Exactly, that could be a problem if you want to have a spawn on top of an open trapdoor above a deep abyss, for example. I think in this case we don't want to "fix" what ain't broken.